filebeat/input/kafka/input.go

```diff
@@ -307,8 +325,29 @@ func (h *groupHandler) ack(message *sarama.ConsumerMessage) {
 
 func (h *groupHandler) ConsumeClaim(sess sarama.ConsumerGroupSession, claim sarama.ConsumerGroupClaim) error {
 	for msg := range claim.Messages() {
-		event := h.createEvent(sess, claim, msg)
-		h.outlet.OnEvent(event)
+		events := h.createEvents(sess, claim, msg)
+		for _, event := range events {
+			h.outlet.OnEvent(event)
+		}
 	}
 	return nil
 }
+
+// parseMultipleMessages will try to split the message into multiple ones based on the group field provided by the configuration
+func (h *groupHandler) parseMultipleMessages(bMessage []byte) []string {
```

Comment:
There should be at least one test exercising the multiple-message code path. Kafka is awkward to unit test but you can probably add something to `kafka_integration_test.go` without too much trouble.